### PR TITLE
Remove setenv

### PR DESCRIPTION
**Pull request by DmitriyMusatkin**
*Issue #:* https://github.com/awslabs/aws-c-io/issues/521

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/s2n/s2n_tls_channel_handler.c [hunk 1]
```diff
@@ -196,7 +229,8 @@ void aws_tls_init_static_state(struct aws_allocator *alloc) {
     }
 
     if (!s_s2n_initialized_externally) {
-        setenv("S2N_DONT_MLOCK", "1", 1);
+        //setenv("S2N_DONT_MLOCK", "1", 1);
```
Comment (graebm) [inline source/s2n/s2n_tls_channel_handler.c:232]: trivial: don't commit commented-out code

File: source/s2n/s2n_tls_channel_handler.c [hunk 2]
```diff
@@ -196,7 +229,8 @@ void aws_tls_init_static_state(struct aws_allocator *alloc) {
...
+        s_override_s2n_mem_functions(alloc);
```
Comment (graebm) [inline source/s2n/s2n_tls_channel_handler.c:233]: debatable: why bother making this a helper function?

File: source/s2n/s2n_tls_channel_handler.c [hunk 3]
```diff
@@ -174,6 +174,39 @@ AWS_IO_API const char *aws_determine_default_pki_ca_file(void) {
...
+    (void)size;
+    aws_mem_release(s_library_allocator, ptr);
+    return S2N_SUCCESS;
+}
+
+static void s_override_s2n_mem_functions(struct aws_allocator *alloc) {
+    if (alloc) {
+        s_library_allocator = alloc;
+    } else {
+        s_library_allocator = aws_default_allocator();
```
Comment (graebm) [inline source/s2n/s2n_tls_channel_handler.c:204]: I'm pretty sure we don't allow NULL to be passed into library-init functions. Just ASSERT on it, don't try to handle the error

File: source/s2n/s2n_tls_channel_handler.c [hunk 4]
```diff
@@ -174,6 +174,39 @@ AWS_IO_API const char *aws_determine_default_pki_ca_file(void) {
...
+    }
+
+    s2n_mem_set_callbacks(s_s2n_mem_init, s_s2n_mem_cleanup, s_s2n_mem_malloc, s_s2n_mem_free);
```
Comment (graebm) [inline source/s2n/s2n_tls_channel_handler.c:207]: this returns an error code, we should check it

File: source/s2n/s2n_tls_channel_handler.c (hunk 3 above)
Comment (DmitriyMusatkin) [reply source/s2n/s2n_tls_channel_handler.c:204]: both s3 and auth have their init written this way, so just copying them here. happy to switch to assert if we disallow NULL

Comment (graebm) [reply source/s2n/s2n_tls_channel_handler.c:204]: aws-c-http would crash, aws-c-cal (on linux) would crash,

**Review by waahm7 — APPROVED** [review]

**Pull request merged**